### ISSUE TITLE
fuse creates extra named fields to hold values of the same type

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2855]
Repro is with Zed commit 2b0b0c9.

While working with a community user on the issues ultimately captured in #2852 and #2853, I observed the following `fuse` behavior I can't easily explain. These queries are being performed against the [Zeek NDJSON zed-sample-data](https://github.com/brimdata/zed-sample-data/tree/main/zeek-ndjson).

Examined in isolation, the `tags` field of the Zeek `http` NDJSON logs only ever have two possible values.

```
$ zq -version
Version: v0.29.0-411-g2b0b0c97

$ zq -z 'count()' http.ndjson.gz 
{count:144034 (uint64)} (=0)

$ zq -z 'count() by tags' http.ndjson.gz 
{tags:["HTTP::URI_SQLI"],count:17 (uint64)} (=0)
{tags:[] (1=([null])),count:144017 (uint64)} (=2)
```

If we use `fuse` on the unshaped NDJSON to try to unite it under a single shape, we see `tags` values get stored across four different named fields `tags`, `tags_2`, `tags_3`, and `tags_4`. This is the only field that `fuse` ends up giving the "append a number to hold the variations" treatment.

```
$ zq -Z 'fuse | count() by typeof(this)' http.ndjson.gz 
{
    typeof: ({_path:string,_write_ts:string,ts:string,uid:string,id.orig_h:string,id.orig_p:int64,id.resp_h:string,id.resp_p:int64,trans_depth:int64,method:string,host:string,uri:string,version:string,user_agent:string,request_body_len:int64,response_body_len:int64,status_code:int64,status_msg:string,tags:[null],resp_fuids:[string],resp_mime_types:[string],referrer:string,orig_fuids:[string],orig_mime_types:[string],origin:string,proxied:[string],resp_filenames:[string],username:string,info_code:int64,info_msg:string,tags_2:[string],tags_3:[string],tags_4:[string]}),
    count: 144034 (uint64)
} (=0)
```

Examining these in isolation:

```
$ zq -z 'fuse | count() by tags' http.ndjson.gz
{tags:[] (0=([null])),count:144017 (uint64)} (=1)
{tags:null,count:17} (1)

$ zq -z 'fuse | count() by tags_2' http.ndjson.gz
{tags_2:null (0=([string])),count:144021 (uint64)} (=1)
{tags_2:["HTTP::URI_SQLI"],count:13} (1)

$ zq -z 'fuse | count() by tags_3' http.ndjson.gz
{tags_3:null (0=([string])),count:144031 (uint64)} (=1)
{tags_3:["HTTP::URI_SQLI"],count:3} (1)

$ zq -z 'fuse | count() by tags_4' http.ndjson.gz
{tags_4:null (0=([string])),count:144033 (uint64)} (=1)
{tags_4:["HTTP::URI_SQLI"],count:1} (1)
```

After having studied the behavior described in #2852, I was expecting no more than two variations on `tags`, one to hold the "array of nulls" values and one to hold the "array of strings" values. So I can't explain the presence of `tags_3` and `tags_4`.

Bottom line: If it turns out this is working as expected, I'd need some help updating the user-facing docs to explain the behavior.

BTW, the symptom seems to have something to do with the presence of all the other fields in the records, since if we isolate just the `tags` values before we `fuse`, now we see the expected behavior similar to what's reported in #2852.

```
$ zq -z 'cut tags | fuse | count() by typeof(this)' http.ndjson.gz 
{typeof:({tags:[null],tags_2:[string]}),count:144034 (uint64)} (=0)
```

**Comment by philrz** [comment]
Verified in Zed commit 8ca4e07.

Repeating the repro steps, we see now that the variations like `tags_2`, `tags3`, etc. are no longer present, and instead `tags` exists as a single field of the `union` type `([null],[string])`.

```
$ zq -Z 'fuse | count() by typeof(this)' http.ndjson.gz
{
    typeof: ({_path:string,_write_ts:string,ts:string,uid:string,id.orig_h:string,id.orig_p:int64,id.resp_h:string,id.resp_p:int64,trans_depth:int64,method:string,host:string,uri:string,version:string,user_agent:string,request_body_len:int64,response_body_len:int64,status_code:int64,status_msg:string,tags:([null],[string]),resp_fuids:[string],resp_mime_types:[string],referrer:string,orig_fuids:[string],orig_mime_types:[string],origin:string,proxied:[string],resp_filenames:[string],username:string,info_code:int64,info_msg:string}),
    count: 144034 (uint64)
} (=0)
```

Thanks @nwt!
